### PR TITLE
e2e test - additional pyrefly coverage

### PR DESCRIPTION
**Pull request by rodrigosf672**
This pull request expands the test coverage for Python diagnostics in the LSP integration tests, aiming to catch potential issues and breakages in diagnostic functionality. It introduces several new test cases for common Python errors and edge cases, and prepares groundwork for future tests on QMD and Notebook files. Additionally, a minor documentation update clarifies a parameter name in the `Problems` page object (`badgeCount`).

Expanded Python diagnostics test coverage:

* Added multiple new tests to `lsp-diagnostics.test.ts` to verify diagnostics for scenarios such as non-existent module imports, missing attributes, calling non-callables, incompatible assignments, wrong return types, list/dict element type mismatches, and type errors with indentation issues. These tests improve robustness and help detect pyrefly-related issues or delays.
* Added skipped tests for QMD and Notebook files, noting current limitations due to pyrefly not supporting `.qmd` and `.ipynb` files. These are placeholders for future diagnostic coverage.

Test infrastructure improvement:

* Imported `path` module in `lsp-diagnostics.test.ts` to support file path operations for future and existing tests.

Documentation update:

* Clarified the parameter name from `problemCount` to `badgeCount` in the `Problems` page object, making the expected badge count meaning more explicit.

@:pyrefly @:web

### CODE REVIEW

File: test/e2e/tests/lsp/lsp-diagnostics.test.ts [hunk 1]
```diff
@@ -64,4 +65,120 @@ test.describe('Diagnostics', {
...
+
+	test('Python - File shows type errors when indentation error is present', async function ({ app, runCommand, sessions, python }) {
+		const { problems, editor } = app.workbench;
+		await runCommand('Python: New File');
+		await editor.type('def greet(name: str) -> str:\n\treturn f"Hello, {name}"\n\n# Type error: passing int instead of str\nresult = greet(123)\n');
+		await problems.expectDiagnosticsToBe({ badgeCount: 6, warningCount: 0, errorCount: 6 });
+	});
+
+	// NOTE[RSF]: Unskip/fix tests for QMD and Notebook files
+	// Currently skipping due to pyrefly not working with .qmd and .ipynb
```
Comment (austin3dickey) [inline test/e2e/tests/lsp/lsp-diagnostics.test.ts:140]: I think I've fixed the ipynb issue in https://github.com/posit-dev/positron/pull/12162. Would you mind merging in the latest main, and unskipping those tests?

For qmd, I think pyrefly is actually working correctly. I see semantic highlighting, hover, completions, etc. But the quarto extension is suppressing diagnostics, because at the top of its virtual doc it includes `# type: ignore`. I found this issue: https://github.com/quarto-dev/quarto/issues/208

Would you mind linking that issue to the skipped qmd tests? And in the future if you add pyrefly tests for qmd files for things like hover/highlighting, those should work.

Comment (juliasilge) [reply test/e2e/tests/lsp/lsp-diagnostics.test.ts:140]: FYI I put diagnostics/squiggles in `.qmd` as my top, top priority for the Quarto source editor for this year: https://github.com/quarto-dev/quarto/issues/897

**Comment by testlabauto** [comment]
Don't forget that this is how you test in CI:
https://github.com/posit-dev/positron/actions/workflows/test-e2e-pyrefly.yml

I ran against your PR and it looks good!

For a refresher, here is what happens: 
https://github.com/posit-dev/positron/blob/main/playwright.config.ts#L46
That ^ skips tests in the lsp directory unless ALLOW_PYREFLY is set. It is only set with the workflow I mentioned above and with the nightly tests.

So, its annoying but using the tag for pyrefly on your PR doesn't do anything.

**Review by testlabauto — APPROVED** [review]
LGTM!
